datadog_checks_base/datadog_checks/base/utils/tracing.py

```diff
@@ -16,6 +17,14 @@
 
 EXCLUDED_MODULES = ['threading']
 
+# During regular continuous tracing we trace only the check's top-level 'run' and
+# 'check' methods. This is because some checks may make 1000s of calls to various
+# internal methods, creating many 1000s of spans, generating excessive overhead.
+# All methods are traced only if exhaustive tracing is enabled. This is generally
+# done only during a single manually triggered check run, during which such overhead
+# is acceptable.
+AGENT_CHECK_DEFAULT_TRACED_METHODS = {'check', 'run', 'warning'}
```

Comment:
Since the execution of a check is always triggered by the run method shouldn't we be fine with just tracing that one?

Reply:
I thought that initially but the problem is that the `run` method handles any exceptions thrown by `check`, which means if we only trace `run` then we won't see those exceptions in the trace.

Reply:
I see